packages/next/telemetry/storage.ts

```diff
@@ -171,8 +177,21 @@ export class Telemetry {
         isRejected: true,
         reason,
       }))
+      // Acts as `Promise#finally` because `catch` transforms the error
+      .then(res => {
+        // Clean up the event to prevent unbounded `Set` growth
+        this.queue.delete(prom)
+        return res
+      })
+
+    // Track this `Promise` so we can flush pending events
+    this.queue.add(prom)
+
+    return prom
   }
 
+  flush = async () => Promise.all(this.queue).catch(() => null)
```

Comment:
```suggestion
  flush = () => Promise.all(this.queue).catch(() => null)
```
We're already returning a `Promise` so it doesn't need to be marked as `async` right?

Reply:
I did this intentionally; just a general practice I follow to prevent the function from flopping if modified (e.g. we add a short-circuit case for empty `queue`).